### PR TITLE
chore(db): add data migration for new entity field subtypes

### PR DESCRIPTION
**Pull request by overbit**
Close: #6645 

### Summary
🔢🔄🛠️

This pull request adds manual data migration scripts to update the Prisma data type properties of entity fields that have the new DecimalNumber and WholeNumber data types in Amplication. These scripts are required to ensure compatibility and consistency between the Amplication and Prisma data models.

> _`DecimalNumber`_
> _and `WholeNumber` migrate_
> _to Prisma types - FLOAT, INT_

### Walkthrough
*  Add manual data migration scripts to support new data types in Amplication ([link](https://github.com/amplication/amplication/pull/6646/files?diff=unified&w=0#diff-fef4b4f90e55625cd0e7d233efafd9f8a8039ad21ea60b1223441d34f777a5d6L1-R8), [link](https://github.com/amplication/amplication/pull/6646/files?diff=unified&w=0#diff-21895e8d15c90b7c55b0330c8f9b361b16cbc0a503be834e188c1beaa3172990R1-R10))
  - Update entity fields with DecimalNumber data type to have FLOAT data type property in `20230809_add_decimal_datatype.sql` ([link](https://github.com/amplication/amplication/pull/6646/files?diff=unified&w=0#diff-fef4b4f90e55625cd0e7d233efafd9f8a8039ad21ea60b1223441d34f777a5d6L1-R8))
  - Update entity fields with WholeNumber data type to have INT data type property in `20230809_add_wholenumber_datatype.sql` ([link](https://github.com/amplication/amplication/pull/6646/files?diff=unified&w=0#diff-21895e8d15c90b7c55b0330c8f9b361b16cbc0a503be834e188c1beaa3172990R1-R10))

## PR Checklist

- [x] Tests for the changes have been added
- [x] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/CODE_OF_CONDUCT.md) file for detailed contributing guidelines.

### CODE REVIEW

**Review by abrl91 — APPROVED** [review]

File: packages/amplication-prisma-db/prisma/manual-data-migrations/20230809_add_decimal_datatype.sql [hunk 1]
```diff
@@ -0,0 +1,9 @@
+-- SELECT ef."name", ef."dataType", ef.properties
+-- FROM "EntityField" as ef
+-- WHERE ef."dataType" = 'DecimalNumber'
+-- AND NOT (ef.properties)::jsonb ? 'dataType';
+
+UPDATE "EntityField"
+SET properties = properties || '{ "dataType": "FLOAT" }'
```
Comment (abrl91) [inline packages/amplication-prisma-db/prisma/manual-data-migrations/20230809_add_decimal_datatype.sql:7]:
```suggestion
SET properties = properties || '{ "databaseFieldType": "FLOAT" }'
```

File: packages/amplication-prisma-db/prisma/manual-data-migrations/20230809_add_wholenumber_datatype.sql [hunk 2]
```diff
@@ -0,0 +1,10 @@
+-- SELECT ef."name", ef."dataType", ef.properties
+-- FROM "EntityField" as ef
+-- WHERE ef."dataType" = 'WholeNumber'
+-- AND NOT (ef.properties)::jsonb ? 'dataType';
+
+UPDATE "EntityField"
+SET properties = properties || '{ "dataType": "INT" }'
```
Comment (abrl91) [inline packages/amplication-prisma-db/prisma/manual-data-migrations/20230809_add_wholenumber_datatype.sql:7]:
```suggestion
SET properties = properties || '{ "databaseFieldType": "INT" }'
```

File: packages/amplication-prisma-db/prisma/manual-data-migrations/20230809_add_wholenumber_datatype.sql [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+WHERE "dataType" = 'WholeNumber'
+AND NOT (properties)::jsonb ? 'dataType';
```
Comment (abrl91) [inline packages/amplication-prisma-db/prisma/manual-data-migrations/20230809_add_wholenumber_datatype.sql:9]:
```suggestion
AND NOT (properties)::jsonb ? 'databaseFieldType';
```

File: packages/amplication-prisma-db/prisma/manual-data-migrations/20230809_add_decimal_datatype.sql [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+WHERE "dataType" = 'DecimalNumber'
+AND NOT (properties)::jsonb ? 'dataType';
```
Comment (abrl91) [inline packages/amplication-prisma-db/prisma/manual-data-migrations/20230809_add_decimal_datatype.sql:9]:
```suggestion
AND NOT (properties)::jsonb ? 'databaseFieldType';
```

File: packages/amplication-prisma-db/prisma/manual-data-migrations/20230809_add_decimal_datatype.sql [hunk 5]
```diff
@@ -0,0 +1,9 @@
+-- SELECT ef."name", ef."dataType", ef.properties
+-- FROM "EntityField" as ef
+-- WHERE ef."dataType" = 'DecimalNumber'
+-- AND NOT (ef.properties)::jsonb ? 'dataType';
```
Comment (abrl91) [inline packages/amplication-prisma-db/prisma/manual-data-migrations/20230809_add_decimal_datatype.sql:4]:
```suggestion
-- AND NOT (ef.properties)::jsonb ? 'databaseFieldType';
```

File: packages/amplication-prisma-db/prisma/manual-data-migrations/20230809_add_wholenumber_datatype.sql [hunk 6]
```diff
@@ -0,0 +1,10 @@
+-- SELECT ef."name", ef."dataType", ef.properties
+-- FROM "EntityField" as ef
+-- WHERE ef."dataType" = 'WholeNumber'
+-- AND NOT (ef.properties)::jsonb ? 'dataType';
```
Comment (abrl91) [inline packages/amplication-prisma-db/prisma/manual-data-migrations/20230809_add_wholenumber_datatype.sql:4]:
```suggestion
-- AND NOT (ef.properties)::jsonb ? 'databaseFieldType';
```

**Comment by mulygottlieb** [comment]
@overbit @abrl91 @levivannoort 
Just FYI so we are all aligned - we decided NOT to run this migration on production. Instead, Amit will add default fallbacks in the code itself to add the `databaseFieldType` property if it's missing in the DB.

A data migration in production is potentially very dangerous and it isn't always easy to roll back from it in case it goes sideways. So data migration should be the last resort only if we have no other choice.

**Pull request merged**